spec/src/modules/assistant.js

```diff
@@ -0,0 +1,264 @@
+
+    if (bundled) {
+      ConstructorIO = window.ConstructorioClient;
+    }
+  });
+
+  afterEach(() => {
+    delete global.CLIENT_VERSION;
+    delete window.CLIENT_VERSION;
+    cleanup();
+  });
+
+  // createAssistantUrl util Tests
+  describe('Test createAssistantUrl', () => {
+
+    it('should throw an error if intent is not provided', () => {
+      expect(() => createAssistantUrl('', defaultParameters, defaultOptions)).throw('intent is a required parameter of type string');
+    });
+
+    it('should throw an error if domain is not provided in parameters', () => {
+      expect(() => createAssistantUrl('testIntent', {}, defaultOptions)).throw('parameters.domain is a required parameter of type string');
+    });
+
+    it('should correctly construct a URL with minimal valid inputs', () => {
+      const intent = 'testIntent';
+      const url = createAssistantUrl(intent, defaultParameters, defaultOptions);
+
+      expect(url).contain('https://assistant.cnstrc.com/v1/intent/');
+      expect(url).contain(`intent/${encodeURIComponentRFC3986(intent)}`);
+
+      const requestedUrlParams = qs.parse(url.split('?')?.[1]);
+
+      expect(requestedUrlParams).to.have.property('key');
+      expect(requestedUrlParams).to.have.property('i');
+      expect(requestedUrlParams).to.have.property('s');
+      expect(requestedUrlParams).to.have.property('c').to.equal(clientVersion);
+      expect(requestedUrlParams).to.have.property('_dt');
+
+    });
+
+    it('should clean and encode parameters correctly', () => {
+      const intentWithSpaces = 'test/ [Intent)';
+      const url = createAssistantUrl(
+        intentWithSpaces,
+        { ...defaultParameters },
+        defaultOptions,
+      );
+
+      // Ensure spaces are encoded and parameters are cleaned
+      expect(url).not.contain(' ');
+      expect(url).contain(encodeURIComponentRFC3986(intentWithSpaces));
+    });
+  });
+
+  // setupEventListeners util Tests
+  describe('Test setupEventListeners', () => {
+    let mockEventSource;
+    let mockStreamController;
+
+    beforeEach(() => {
+      mockEventSource = {
+        addEventListener: sinon.stub(),
+        close: sinon.stub(),
+      };
+
+      mockStreamController = {
+        enqueue: sinon.stub(),
+        close: sinon.stub(),
+        error: sinon.stub(),
+      };
+    });
+
+    afterEach(() => {
+      sinon.restore(); // Restore all mocks
+    });
+
+    it('should set up event listeners for all event types', () => {
+      const eventTypes = Assistant.EventTypes;
+
+      setupEventListeners(mockEventSource, mockStreamController, eventTypes);
+      Object.values(Assistant.EventTypes).forEach((event) => {
+        expect(mockEventSource.addEventListener.calledWith(event)).to.be.true;
+      });
+    });
+
+    it('should enqueue event data into the stream', (done) => {
+      const eventTypes = Assistant.EventTypes;
+      const eventType = Assistant.EventTypes.SEARCH_RESULT;
```

Comment:
Not a big deal and something I noticed after I asked for this change, but we could probably just use `Assistant.EventTypes` and `Assistent.EventTypes.<event>` without needing to set them to variables. This would make the tests a bit more verbose since `eventType` isn't specific.

(this would apply to the other places we are using this)